### PR TITLE
DOC-4214: Update cache configuration properties documentation

### PR DESCRIPTION
**Pull request by mzorai**

### CODE REVIEW

File: docs/Configuration.rst [hunk 1]
```diff
@@ -2908,46 +2908,43 @@ These Portal caches can be overridden in
...
-    #  - Standalone (live time in seconds)
-    exo.cache.portal.mop.maxsize=5000
-    exo.cache.portal.mop.livetime=600
-    #  - Cluster (expiration in milliseconds)
-    exo.cache.portal.mop.maxnodes=5000
-    exo.cache.portal.mop.expiration=600000
+    exo.cache.portal.mop.MaxNodes=1400
+    exo.cache.portal.mop.TimeToLive=86400
+    # For Cluster mode
+    exo.cache.portal.mop.strategy=LIRS
```
Comment (thomasdelhomenie) [inline docs/Configuration.rst:2918]: "strategy" is not only for cluster mode, should be moved above.

File: docs/Configuration.rst [hunk 2]
```diff
@@ -3087,23 +3084,66 @@ loaded from cache rather than the database or the file system.
...
+
+-  :ref:`Web Notification Cache <WebNotificationCache>`
+
+-  :ref:`Web Notifications Cache <WebNotificationsCache>`
+
+-  :ref:`User State Service <UserStateService>`
+
+-  :ref:`User Setting Service <UserSettingService>`
+
+These Common caches can be overridden in
```
Comment (thomasdelhomenie) [inline docs/Configuration.rst]: The word "Common" does not make a lot of sense here IMO. We should instead talk about Settings, Notifications and User State caches.

File: docs/Configuration.rst [hunk 3]
```diff
@@ -3087,23 +3084,66 @@ loaded from cache rather than the database or the file system.
...
+    exo.cache.commons.SettingService.TimeToLive=86400
+    # For cluster mode
+    exo.cache.commons.SettingService=replication
+    exo.cache.commons.SettingService.strategy=LIRS
+    
+	# Commons Cache Configuration - Web Notification Count
+	exo.cache.commons.WebNotificationCountCache.MaxNodes=1000
+	exo.cache.commons.WebNotificationCountCache.TimeToLive=86400
+	exo.cache.commons.WebNotificationCountCache.strategy=LIRS
+	exo.cache.commons.WebNotificationCountCache.cacheMode=asyncReplication
```
Comment (thomasdelhomenie) [inline docs/Configuration.rst]: Is the extra indentation intended ?

**Review by thomasdelhomenie — CHANGES_REQUESTED** [review]

File: docs/Configuration.rst (hunk 3 above)
Comment (mzorai) [reply docs/Configuration.rst]: nope, fixed now thanks!

File: docs/Configuration.rst [hunk 4]
```diff
@@ -2908,46 +2908,44 @@ These Portal caches can be overridden in
...
-    exo.cache.portal.mop.maxsize=5000
-    exo.cache.portal.mop.livetime=600
-    #  - Cluster (expiration in milliseconds)
-    exo.cache.portal.mop.maxnodes=5000
-    exo.cache.portal.mop.expiration=600000
+    exo.cache.portal.mop.MaxNodes=1400
+    exo.cache.portal.mop.TimeToLive=86400
+    exo.cache.portal.mop.strategy=LIRS
+    # For Cluster mode
+
```
Comment (thomasdelhomenie) [inline docs/Configuration.rst]: Useless blank line

File: docs/Configuration.rst [hunk 5]
```diff
@@ -2908,46 +2908,44 @@ These Portal caches can be overridden in
...
-    #  - Standalone (live time in seconds)
-    exo.cache.portal.navigation.maxsize=5000
-    exo.cache.portal.navigation.livetime=600
-    #  - Cluster (expiration in milliseconds)
-    exo.cache.portal.navigation.maxnodes=5000
-    exo.cache.portal.navigation.expiration=600000
+    exo.cache.portal.navigation.MaxNodes=700
+    exo.cache.portal.navigation.TimeToLive=86400
+    # For Cluster mode
+    exo.cache.portal.navigation.strategy=LIRS
```
Comment (thomasdelhomenie) [inline docs/Configuration.rst:2925]: "strategy" to move above (in standalone mode) for all caches.

File: docs/Configuration.rst [hunk 6]
```diff
@@ -2908,46 +2908,44 @@ These Portal caches can be overridden in
...
+    exo.cache.portal.mop.cacheMode=replication
 
     # Portal Cache Configuration - Navigation Service
-    #  - Standalone (live time in seconds)
-    exo.cache.portal.navigation.maxsize=5000
-    exo.cache.portal.navigation.livetime=600
-    #  - Cluster (expiration in milliseconds)
-    exo.cache.portal.navigation.maxnodes=5000
-    exo.cache.portal.navigation.expiration=600000
+    exo.cache.portal.navigation.MaxNodes=700
```
Comment (thomasdelhomenie) [inline docs/Configuration.rst]: You used the values defined in exo-sample.properties right ? Shouldn't we use the default value instead (defined in https://github.com/exodev/platform/blob/develop/extension/config/src/main/resources/conf/platform/configuration.properties) ? We need these default values somewhere IMO.

**Review by thomasdelhomenie — CHANGES_REQUESTED** [review]

File: docs/Configuration.rst (hunk 6 above)
Comment (mzorai) [reply docs/Configuration.rst]: Yes i used values of exo-sample.properties file OK i will change them

Comment (mzorai) [reply docs/Configuration.rst]: Values updated

File: docs/Configuration.rst [hunk 7]
```diff
@@ -2882,6 +2882,8 @@ explanation of properties used for eXo Platform caches.
 		-  :ref:`Cache Levels <MonitoringGadget.CacheLevels>` in User Guide.
 
 		-  :ref:`Cache management view <Management.ManagementViews.CacheManagementView>`.
+		
+You can have the full list of different caches and their values `here <https://github.com/exodev/platform/blob/develop/extension/config/src/main/resources/conf/platform/configuration.properties#L136>`__
```
Comment (thomasdelhomenie) [inline docs/Configuration.rst]: No need to link to the internal configuration properties file IMO.
We just have to tell that the below properties are all the cache configuration properties with their default value and that they must be changed to tune eXoPlatform.

File: docs/Configuration.rst [hunk 8]
```diff
@@ -2908,46 +2910,43 @@ These Portal caches can be overridden in
...
-    exo.cache.portal.mop.maxsize=5000
-    exo.cache.portal.mop.livetime=600
-    #  - Cluster (expiration in milliseconds)
-    exo.cache.portal.mop.maxnodes=5000
-    exo.cache.portal.mop.expiration=600000
+    exo.cache.portal.mop.MaxNodes=5000
+    exo.cache.portal.mop.TimeToLive=-1
+    exo.cache.portal.mop.strategy=LIRS
+    # For Cluster mode
+    exo.cache.portal.mop.cacheMode=replication
```
Comment (thomasdelhomenie) [inline docs/Configuration.rst]: The cache mode of caches config are not good, see https://github.com/exodev/platform/blob/develop/extension/config/src/main/resources/conf/platform/configuration.properties#L689 for the good default values.

**Review by thomasdelhomenie — CHANGES_REQUESTED** [review]

File: docs/Configuration.rst (hunk 2 above)
Comment (thomasdelhomenie) [reply docs/Configuration.rst]: @mzorai this comment has not been considered

File: docs/Configuration.rst (hunk 7 above)
Comment (mzorai) [reply docs/Configuration.rst]: OK

File: docs/Configuration.rst (hunk 8 above)
Comment (mzorai) [reply docs/Configuration.rst]: (y)

**Review by thomasdelhomenie — APPROVED** [review]

**Pull request merged**